### DISCUSSION TITLE
Building from source - easy! (Fedora 36)

### DISCUSSION THREAD

**Discussion by KarlLevik** [neovide/neovide#1357] [General]
Just wanted to say I recently decided to try and build from source since the snap image still doesn't work on Fedora, and I was pleasantly surprised at how easy it was. These are the steps I followed on a new install of Fedora 36:

```bash
dnf install git cargo gcc-c++ freetype-devel fontconfig-devel
git clone "https://github.com/neovide/neovide"
cd neovide
cargo build --release
```

... and then copied `target/release/neovide` to somewhere in my PATH. 

But the `README.md` build instructions for building from source are somewhat longer and different. So I was wondering if I'm missing functionality / features by not installing all the packages listed in `README.md`, and not using the Rust install script from `https://sh.rustup.rs`? 

The build completed without any errors that I could see, so does that mean it's all good?

**Comment by MultisampledNight** [comment]
Probably everything's fine. Only problem would occur if Neovide would use new functionality introduced by a recent Rust version, which your distro might not have in its repositories yet. And the nice thing about rustup is that it works _everywhere,_ if a distro would have a very complicated setup process for installing packages, we can always refer to rustup as an easier option.
